lib/linalg/BasisReader.h

```diff
@@ -197,6 +200,11 @@ class BasisReader {
      *
      * @brief Returns the dimension of the system on this processor.
      *
+     * @param[in] kind  The target matrix of which dimension (row) is returned.
```

Comment:
This comment is not clear to me. Would this be better? `Type of matrix whose row-dimension is returned.`